src/main/scala/io/qbeast/core/model/QbeastSnapshot.scala

```diff
@@ -29,6 +31,18 @@ trait QbeastSnapshot {
    */
   def isInitial: Boolean
 
+  /**
+   * The current table description.
+   * @return
+   */
+  def loadDescription: String
+
+  /**
+   * The current table properties of the snapshot.
+   * @return
+   */
+  def loadProperties: mutable.Map[String, String]
```

Comment:
I see that the output is later updated with more key-value pairs, but isn't it better to return an immutable Map?
The updates `QbeastTableImpl.properties()` can initialize a map from `loadProperties` instead of changing it.